core/src/main/java/dev/morphia/aggregation/Aggregation.java

```diff
@@ -497,20 +478,23 @@ default Aggregation<T> set(AddFields fields) {
      * aggregation pipeline.
      *
      * @return this
-     * @since 2.3
      * @aggregation.expression $changeStream
+     * @since 2.3
      */
-    Aggregation changeStream();
+    Aggregation<T> changeStream();
 
     /**
      * Returns a Change Stream cursor on a collection, a database, or an entire cluster. Must be used as the first stage in an
      * aggregation pipeline.
      *
      * @param stream the options to apply to the stage
-     *
      * @return this
-     * @since 2.3
      * @aggregation.expression $changeStream
+     * @since 2.3
      */
-    Aggregation changeStream(ChangeStream stream);
+    Aggregation<T> changeStream(ChangeStream stream);
+
+    List<Stage> getStates();
```

Comment:
This method should not be on the public interface.  I understand *why* it's there.  You want to test.  But you should downcast in your test.  There's no reason for user code to call this as it's an internal implementation detail.